### DISCUSSION TITLE
Inserting a video file into a post

### DISCUSSION THREAD

**Discussion by yfarkash** [GetPublii/Publii#1784] [Help]
Hi,

Is there a way to insert a video file into a post?
Not an embed code or a link, but the actual video file?

When adding pictures to a post I insert the actual files from my computer, and they are uploaded to my server when I publish the post.

I would like to do the same with video.  I prefer to just insert the file from my computer into the post, instead of uploading to a hosting site and getting the link.

Is there a way to do that?

Thanks!

**Comment by candidexmedia** [comment]
Besides animated webp files, I don't think it's possible to directly upload a video file.

**Reply by michaelfluegge** [reply]
The documentation states: "_video media elements are inserted into a post by providing the source URL or Embed code_"
https://getpublii.com/dev/default-publii-classes-for-using-with-css/#video

**Reply by candidexmedia** [reply]
Source URL, based on my experience with the different editors, typically means the link to the file hosted on another server. These videos would be displayed inside of the browser's default HTML5 player (unless a third party media player JS is loaded).

Embed codes: YouTube Vimeo, etc. and other sites with their own players displayed in an iframe.

It sounds like OP wants to upload a video to the Publii site like you would an image...

**Comment by gpsblues** [comment]
Hi @yfarkash,

At the moment, I don’t think this is directly possible. However, if your setup follows a consistent page structure (e.g., the video always appears in the same location), you could create a custom post template with an upload control.

Although the upload control is intended for images, you can upload a video file instead. Then, using the reference to the uploaded video, you could construct the `<video>` tag at the specific location within the post without writing extra code.

Here’s a rough, non-functional example to give you an idea of how this could work:

In `config.json`:

```
"postConfig": [
    {
        "name": "videoUpload",
        "label": "Video upload",
        "note": "Upload a .mp4 video",
        "value": "",
        "type": "image"
    },
    ...
]
```
In `post.hbs`:
```
<div class="wrapper post__entry">    
    {{#if @config.post.videoUpload}}
       <video width="100%" height="500" controls>
          <source src="{{@config.post.videoUpload}}" type="video/mp4">
       </video>
    {{/if}}       
    {{{text}}}            
</div>
```

This is a basic setup, but it could serve as a starting point.
